### PR TITLE
StructVisitor cleanup

### PR DESCRIPTION
**Pull request by dfed**
Followup to #56. See inline comments for discussion.

### CODE REVIEW

File: Sources/SwiftInspectorVisitors/StructVisitor.swift [hunk 1]
```diff
@@ -100,10 +100,15 @@ public final class StructVisitor: SyntaxVisitor {
       // TODO: Utilize enum visitor to find inner enum information, utilizing parent information from structInfo
     } else {
       // We've encountered a class declaration before encountering a struct declaration. Something is wrong.
-      assertionFailure("Encountered a top-level class. This is a usage error: a single StructVisitor instance should start walking only over a node of type `StructDeclSyntax`")
+      assertionFailure("Encountered a top-level enum. This is a usage error: a single StructVisitor instance should start walking only over a node of type `StructDeclSyntax`")
```
Comment (dfed) [inline Sources/SwiftInspectorVisitors/StructVisitor.swift:103]: copy/paste error!

File: Sources/SwiftInspectorVisitors/StructVisitor.swift [hunk 2]
```diff
@@ -100,10 +100,15 @@ public final class StructVisitor: SyntaxVisitor {
...
     }
     return .skipChildren
+  }
 
+  public override func visit(_ node: ProtocolDeclSyntax) -> SyntaxVisitorContinueKind {
```
Comment (dfed) [inline Sources/SwiftInspectorVisitors/StructVisitor.swift:108]: we should also assert if we come across a protocol! forgot about this one.

File: Sources/SwiftInspectorVisitors/Tests/StructVisitorSpec.swift [hunk 3]
```diff
@@ -49,10 +49,10 @@ final class StructVisitorSpec: QuickSpec {
               self.sut,
               overContent: content)
 
-            expect(self.sut.structs.first) == StructInfo(
-              name: "SomeStruct",
-              inheritsFromTypes: [],
-              parentTypeName: nil)
```
Comment (dfed) [inline Sources/SwiftInspectorVisitors/Tests/StructVisitorSpec.swift:55]: Here and below I'm trying to follow the advice given in https://github.com/fdiaz/SwiftInspector/pull/56#discussion_r566425656

File: Sources/SwiftInspectorVisitors/Tests/StructVisitorSpec.swift [hunk 4]
```diff
@@ -333,6 +333,22 @@ final class StructVisitorSpec: QuickSpec {
             .to(throwAssertion())
         }
       }
+
+      context("visiting a code block with a protocol declaration") {
```
Comment (dfed) [inline Sources/SwiftInspectorVisitors/Tests/StructVisitorSpec.swift:337]: added a test for new functionality

**Review by fdiaz — APPROVED** [review]

**Pull request merged**